hardware/Meteorologisk.cpp

```diff
@@ -48,22 +48,90 @@ std::string ReadFile(std::string filename)
 }
 #endif
 
+#define Meteorologisk_URL "https://api.met.no/weatherapi/locationforecast/2.0/complete.json"
+#define Meteorologisk_forecast_URL "https://darksky.net/#/f/"
+
 CMeteorologisk::CMeteorologisk(const int ID, const std::string &Location) :
 	m_Location(Location)
 {
 	m_HwdID=ID;
-	Init();
 }
 
 void CMeteorologisk::Init()
 {
+	std::stringstream sURL;
+	m_Lat = 1;
+	m_Lon = 1;
+
+	// Try the hardware location first, then fallback on the global settings
+	if(!m_Location.empty())
+	{
+		std::vector<std::string> strarray;
+		if(m_Location != "0")
+		{
+			StringSplit(m_Location, ",", strarray);
+			if(strarray.size() == 2)
+			{
+				m_Lat = strtod(strarray[0].c_str(),nullptr);
+				m_Lon = strtod(strarray[1].c_str(),nullptr);
+			}
+			else
+			{
+				StringSplit(m_Location, ";", strarray);
+				if(strarray.size() == 2)
+				{
+					m_Lat = strtod(strarray[0].c_str(),nullptr);
+					m_Lon = strtod(strarray[1].c_str(),nullptr);
+					Log(LOG_STATUS, "Found Location with semicolon as seperator instead of comma.");
+				}
+			}
+		}
+		else
+		{
+			int nValue;
+			std::string sValue;
+			if (!m_sql.GetPreferencesVar("Location", nValue, sValue))
+			{
+				Log(LOG_ERROR, "Invalid Location found in Settings! (Check your Latitude/Longitude!)");
+			}
+			else
+			{
+				StringSplit(sValue, ";", strarray);
+				if(strarray.size() == 2)
+				{
+					m_Lat = strtod(strarray[0].c_str(),nullptr);
+					m_Lon = strtod(strarray[1].c_str(),nullptr);
+					Log(LOG_STATUS, "Using Domoticz default location from Settings as Location!");
+				}
+			}
+		}
+	}
+
+	if(!(m_Lat == 1 || m_Lon == 1))
+	{
+		sURL << Meteorologisk_URL << "?lat=" << m_Lat << "&lon=" << m_Lon;
+		m_URL = sURL.str();
+	}
+
+	std::string szLoc = CURLEncode::URLEncode(m_Location);
+	sURL.str(std::string());
+	sURL.clear();
+	sURL << Meteorologisk_forecast_URL << szLoc;
+	m_ForecastURL = sURL.str();
+	Debug(DEBUG_HARDWARE, "Meteorologisk: Set Forecast URL to: %s", m_ForecastURL.c_str());
 }
 
 bool CMeteorologisk::StartHardware()
 {
+	Init();
+
+	if(m_URL.empty()) {
+		Log(LOG_STATUS, "Unable to start due to missing or incorrect Location parameters (Latitude, Longitude)!");
+		return false;
+	}
```

Comment:
@gizmocuz , this check prevents the hardware module to actually start its thread because of missing parameters.

But the Domoticz watchdog will report 'unresponsive' as it still expects that a thread should be running to at least update the heartbeat I guess?

Reply:
That's correct. You could add this line at the top of GetMeterDetails. It will log this every 5 minutes but that's fine

Reply:
Done